java/UPDATING.md

```diff
@@ -26,58 +26,32 @@ Test the generated code:
+Note that we are using GitHub actions to publish new releases. These are the [steps](https://dzone.com/articles/how-to-publish-artifacts-to-maven-central) that were followed for publishing the `gtfs-realtime-bindings` library on `ossrh` repository using GitHub actions.
 
-###### 1 - Create an account on [Bintray](https://bintray.com/).
-###### 2 - Setup your pom.xml
-
-We need to specify the URL from which to distribute your project. 
-```
-  <distributionManagement>
-    <repository>
-      <id>jCenter</id>
-      <url>https://api.bintray.com/maven/mobilitydata/mobilitydata-mvn-repo/gtfs-realtime-bindings/;publish=1</url>
-    </repository>
-  </distributionManagement>
-```
-
-###### 3 - Setup your setting.xml
-We need to provide Bintray username and API Key to the Maven `settings.xml` file.  This may be under your Maven installation directory (e.g., `C:\Program Files\Apache Software Foundation\apache-maven-3.2.5\conf`).
-
-```
-<server>
-  <id>jCenter</id>
-  <username>yourjcenteraccount</username>
-  <password>***youraccount-secret-api-key***</password>
-</server>
-```
-
-To sign the application (required for publishing to Maven Central) we use Bintray's automated signing using their internal key as discussed [here](https://www.jfrog.com/confluence/display/BT/Managing+Uploaded+Content#ManagingUploadedContent-SigningwiththeBintrayKey). So, no certificate setup for signing is required.
+To sign the application (required for publishing to Maven Central) we use a GPG key as described [here](https://dzone.com/articles/how-to-publish-artifacts-to-maven-central#:~:text=%3C/servers%3E-,GPG%20Setup,-You%E2%80%99ll%20have%20to).
 
 #### Every release
 
-1. After you've committed any changes, run `mvn release:prepare -Dresume=false`.  This will automatically bump the SNAPSHOT version number to the release version, commit this change, and then bump the version number again to the new SNAPSHOT version for the next development cycle and commit this change too. After running `mvn release:prepare` you'll be prompted to enter these version numbers as well as the tag release name - use the following values, where X is the new version to be released (e.g., if current version is `0.0.1-SNAPSHOT`, release would be `0.0.1`):
+1. Create a new branch from the master branch where you will commit your changes.
+   `git checkout -b your/new/branch`
+
+2. After you've committed any changes, run `mvn release:prepare -Dresume=false`.  This will automatically bump the SNAPSHOT version number to the release version, commit this change, and then bump the version number again to the new SNAPSHOT version for the next development cycle and commit this change too. After running `mvn release:prepare` you'll be prompted to enter these version numbers as well as the tag release name - use the following values, where X is the new version to be released (e.g., if current version is `0.0.1-SNAPSHOT`, release would be `0.0.1`):
```

Comment:
I believe these steps will change now that we have the `mvn deploy` set up to run on PR merge to main branch. I don't think we can use the `mvn release` tasks anymore as currently specified in these docs because the end result would be a merge of the PR to the main branch with a SNAPSHOT version in the pom.xml. I would expect the GitHub Action to then just deploy the new snapshot but not the release that was created in the same PR as the snapshot (because the squashed commit will only contain the final diff of the PR, which is the next snapshot version).

The new workflow will probably look like:
1. Do commit to change to release version
2. Open PR with this release version and merge it
3. Do commit to next snapshot version
4. Open PR with this snapshot version and merge it

Thinking more, I think we might be able to still use `mvn release`, but we wouldn't be able to squash the commits when merging the PR. I believe if each of the commits is merged to the main branch separately, the GitHub Action should still run for each and therefor deploy both the new release and new snapshot. That might be more error prone, though, as it goes against the general PR rules of always preferring squash to merge commits. So maybe it's best to use the manual workflow I listed above...

For now, though, I'd suggest we just replace this whole "Every release" section with a "TODO" to get this PR merged, and then we can tackle the release workflow in a new PR with a new release.

Reply:
It makes sense. Thank you!